test/test_queue_monkeypatch.py

```diff
@@ -0,0 +1,29 @@
+from __future__ import absolute_import
+
+import unittest
+
+import urllib3
+from urllib3.exceptions import EmptyPoolError
+import Queue
+
+class BadError(Exception):
+    """
+    This should not be raised.
+    """
+    pass
+
+Queue.Empty = BadError
```

Comment:
So this monkeypatch needs to be temporary. We have the `mock` module as a test dependency, so we probably want to use `with mock.patch('Queue.Empty', BadError)` in the test instead, to make sure we don't accidentally break other tests.